### DISCUSSION TITLE
How does the GitHub Actions Cache size limit work for Pull Requests?

### DISCUSSION THREAD

**Discussion by lhotari** [community/community#25242] [Actions]
<p>I have read about the update that the <a href="https://github.blog/changelog/2021-11-23-github-actions-cache-size-is-now-increased-to-10gb-per-repository/" rel="noopener nofollow ugc">GitHub Actions Cache size is limited to 10GB</a>.</p>
<p>How does the GitHub Actions Cache size limit work for Pull Requests?</p>
<p>Is the quota for the cache size limit per-user repository (forked repository, where the pull request branch is contained) or for the repository where the pull request is created?</p>
<p>Because of security reasons, a pull request build cannot update the upstream repository’s GitHub Action cache. When the cache gets updated, I would assume that the cache gets stored in the cache associated with the user’s branch and forked repository and thus consumes the quota of the forked repository instead of the upstream repository quota.</p>
<p>Is this the correct understanding?</p>
<p>It would be great to have an explanation in the cache documentation at <a href="https://docs.github.com/en/actions/using-workflows/caching-dependencies-to-speed-up-workflows#usage-limits-and-eviction-policy" class="inline-onebox" rel="noopener nofollow ugc">Caching dependencies to speed up workflows - GitHub Docs</a> .</p>

**Comment by lhotari** [comment]
<p>I found some discussion about the behavior of GitHub Actions Cache with Pull Requests in <a href="https://github.com/actions/cache/issues/79" class="inline-onebox" rel="noopener nofollow ugc">Same cache key not shared between different pull requests · Issue #79 · actions/cache · GitHub</a> . However, there’s no answer to my question there.</p>

**Comment by lhotari** [comment]
<p><a class="mention" href="/dhadka">@dhadka</a> How does the GitHub Actions Cache size limit work for Pull Requests? Would you be able to answer this question? Is the quota of the forked repository used or does it consume the quota of the upstream repository where the Pull Requests is created?</p>

**Comment by lhotari** [comment]
<p>I also found this issue report: <a href="https://github.com/actions/cache/issues/534" class="inline-onebox" rel="noopener nofollow ugc">Pull-Requests evicting cache from master · Issue #534 · actions/cache · GitHub</a> .</p>

**Comment by lhotari** [comment]
<p>I asked the same question in actions/cache discussion: <a href="https://github.com/actions/cache/discussions/776" class="inline-onebox" rel="noopener nofollow ugc">How does the GitHub Actions Cache size limit work for Pull Requests? · Discussion #776 · actions/cache · GitHub</a></p>

**Comment by dhadka** [comment, marked as answer]
<p>It will write to the cache in whichever repo is running the workflow.  To see which it is, find which repo (fork or upstream) that has the run in the Actions tab.  Here’s what I saw with some testing:</p>
<ol>
<li>
<p>If you make changes within a fork (e.g., a PR merging from <code>myfork:mybranch</code> → <code>myfork:main</code>), the Action runs within the fork and the cache says within the fork.</p>
</li>
<li>
<p>If you send a PR to the upstream repo (e.g., <code>myfork:mybranch</code> → <code>upstream:main</code>), then the Action runs within the upstream repo and consequently the cache is created within the upstream repo.</p>
</li>
</ol>
<p>In case there are concerns about case <span class="hashtag">#2</span>, note that a maintainer in the upstream repo must first approve the PR before the workflow will run - <a href="https://docs.github.com/en/actions/managing-workflow-runs/approving-workflow-runs-from-public-forks" class="inline-onebox" rel="noopener nofollow ugc">Approving workflow runs from public forks - GitHub Docs</a>.</p>
